OntologyMapper/src/OntologyMapper/ResourceFileOntologyMappingLoader.cs

```diff
@@ -0,0 +1,63 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="ResourceFileOntologyMappingLoader.cs" company="Microsoft">
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Microsoft.SmartFacilities.OntologyMapper
+{
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+    using System.Reflection;
+
+    public class ResourceFileOntologyMappingLoader : IOntologyMappingLoader
+    {
+        private readonly ILogger logger;
+        private readonly string resourcePath = string.Empty;
+
+        public ResourceFileOntologyMappingLoader(ILogger logger, string resourcePath)
+        {
+            if (string.IsNullOrEmpty(resourcePath))
+            {
+                throw new ArgumentNullException(nameof(resourcePath));
+            }
+
+            this.logger = logger;
+            this.resourcePath = resourcePath;
+        }
+
+        public OntologyMapping LoadOntologyMapping()
+        {
+            logger.LogInformation("Loading Ontology Mapping file: {fileName}", resourcePath);
+
+            var assembly = Assembly.GetExecutingAssembly();
+            var resourceName = assembly.GetManifestResourceNames().Single(str => str.EndsWith(resourcePath));
+
+            using (Stream? stream = assembly.GetManifestResourceStream(resourceName))
+            {
+                if (stream != null)
+                {
+                    using (StreamReader reader = new StreamReader(stream))
+                    {
+                        string result = reader.ReadToEnd();
+                        var mappings = JsonConvert.DeserializeObject<OntologyMapping>(result);
+
+                        if (mappings != null)
+                        {
+                            return mappings;
+                        }
+                        else
+                        {
+                            var error = $"Mappings file '{resourcePath}' is empty or poorly formed.";
```

Comment:
Since these mappings are coming from an embedded resource we should probably have a test to validated them before being checked in, similar to the environment definition files from out last team so bad ones dont get checked in.

Reply:
In progress

Reply:
The issue is also having a copy of the valid model in the repo, which would be even more problematic

Reply:
You could validate that the mapping files all follow the schema so that even if exact values are wrong, at least the file would load? Could be a compromise between making sure the mappings are valid and the making sure the mappings aren't poorly formed

Reply:
Since the mapping jsons are also a key, we'd probably need assurance that they are project unique and someone didn't add two versions of the same name despite being in different folders